### PR TITLE
chore(cli): plumb token header and token prefix oAuth

### PR DESCRIPTION
**Pull request by RohinBhargava**
@coderabbitai summary

### CODE REVIEW

File: packages/cli/cli/versions.yml [hunk 1]
```diff
@@ -1,3 +1,11 @@
+- changelogEntry:
+  - summary: |
+        Docs configuration now lists all the top level roles, and OAuth Client Credential
+        configuration is now more expressive.
```
Comment (dsinghvi) [inline packages/cli/cli/versions.yml:4]: @RohinBhargava can you add a yaml example of how to set the field? 

And do you mind adding a section at the bottom of https://buildwithfern.com/learn/api-definition/fern/authentication#oauth-client-credentials to explain how to configure the header

Comment (RohinBhargava) [reply packages/cli/cli/versions.yml:4]: will do!

File: packages/cli/register/src/ir-to-fdr-converter/convertAuth.ts [hunk 2]
```diff
@@ -48,7 +48,9 @@ export function convertAuth(
                                   endpointId: FdrCjsSdk.EndpointId(
                                       scheme.configuration.tokenEndpoint.endpointReference.endpointId
                                   ),
-                                  accessTokenLocator: FdrCjsSdk.JqString(tokenPath)
+                                  accessTokenLocator: FdrCjsSdk.JqString(tokenPath),
+                                  headerName: scheme.configuration.tokenHeader,
+                                  tokenPrefix: scheme.configuration.tokenPrefix
```
Comment (dsinghvi) [inline packages/cli/register/src/ir-to-fdr-converter/convertAuth.ts:53]: Can you include a fixture here that tests this code path https://github.com/fern-api/fern/tree/main/packages/cli/register/src/ir-to-fdr-converter/__test__/fixtures (probably one specifically for oauth with custom header)

File: fern/apis/fern-definition/definition/auth.yml [hunk 3]
```diff
@@ -86,6 +86,7 @@ types:
       client-id-env: optional<string>
       client-secret-env: optional<string>
       token-prefix: optional<string>
+      token-header: optional<string>
```
Comment (dsinghvi) [inline fern/apis/fern-definition/definition/auth.yml:89]: nit: do you mind adding a `docs` here, once we fix json schema generation to include doc strings that will make the autocomplete nicer for users as well

File: fern/pages/api-definition/fern-definition/auth.mdx [hunk 4]
```diff
@@ -205,4 +205,7 @@ If the `expires-in` property is set, the generated OAuth token provider will aut
 Otherwise, it's assumed that the access token is valid indefinitely.
 
 With this, all of the OAuth logic happens automatically in the generated SDKs. As long as you configure these settings, your
-client will automatically retrieve an access token and refresh it as needed.
\ No newline at end of file
+client will automatically retrieve an access token and refresh it as needed.
+
+When using the docs playground, `token-header` and `token-prefix` can optionally be set to customize the header key name and 
+header value prefix, to match the expected format of the API auth scheme.
```
Comment (dsinghvi) [inline fern/pages/api-definition/fern-definition/auth.mdx:211]: nit: can we add a yaml example below of setting those

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**